dnf-behave-tests/features/steps/cmd.py

```diff
@@ -41,6 +42,33 @@ def when_I_execute_dnf_with_args(context, args):
     context.cmd_exitcode, context.cmd_stdout, context.cmd_stderr = run(cmd, shell=True)
 
 
+@behave.step("I execute \"{binary}\" with args \"{args}\" in directory \"{cwd}\"")
+def when_i_execute_binary_with_args(context, binary, args, cwd):
+    cwd = os.path.join(context.dnf.installroot, cwd.lstrip('/'))
+    cmd = binary
+    cmd += " " + args.format(context=context)
+    context.cmd = cmd
+    context.cmd_exitcode, context.cmd_stdout, context.cmd_stderr = run(cmd, shell=True, cwd=cwd)
+
+
+@behave.step("I execute createrepo_c with args \"{args}\" in directory \"{cwd}\"")
+def when_i_execute_createrepo_c_with_args(context, args, cwd):
+    when_i_execute_binary_with_args(context, "createrepo_c", args, cwd)
+
+
+@behave.step("I execute mergerepo_c with args \"{args}\" in directory \"{cwd}\"")
+def when_i_execute_mergerepo_c_with_args(context, args, cwd):
+    when_i_execute_binary_with_args(context, "mergerepo_c", args, cwd)
+
+
+@behave.step("I execute modifyrepo_c with args \"{args}\" in directory \"{cwd}\"")
+def when_i_execute_modifyrepo_c_with_args(context, args, cwd):
+    when_i_execute_binary_with_args(context, "modifyrepo_c", args, cwd)
+
+@behave.step("I execute sqliterepo_c with args \"{args}\" in directory \"{cwd}\"")
+def when_i_execute_sqliterepo_c_with_args(context, args, cwd):
+    when_i_execute_binary_with_args(context, "sqliterepo_c", args, cwd)
+
```

Comment:
I started to use the generic step at first but I didn't like the quotes around the name of the binary in each step. Therefore I created these thin wrappers since there is only four of them I though its fine and it simplified reading of the steps a bit for me (one less arguments to worry about).